lib/plugins/aws/invokeLocal/index.js

```diff
@@ -150,41 +152,55 @@ class AwsInvokeLocal {
-    const lambdaName = this.options.functionObj.name;
-    const memorySize =
-      Number(this.options.functionObj.memorySize) ||
-      Number(this.serverless.service.provider.memorySize) ||
-      1024;
-
-    const lambdaDefaultEnvVars = {
-      LANG: 'en_US.UTF-8',
-      LD_LIBRARY_PATH:
-        '/usr/local/lib64/node-v4.3.x/lib:/lib64:/usr/lib64:/var/runtime:/var/runtime/lib:/var/task:/var/task/lib', // eslint-disable-line max-len
-      LAMBDA_TASK_ROOT: '/var/task',
-      LAMBDA_RUNTIME_DIR: '/var/runtime',
-      AWS_REGION: this.provider.getRegion(),
-      AWS_DEFAULT_REGION: this.provider.getRegion(),
-      AWS_LAMBDA_LOG_GROUP_NAME: this.provider.naming.getLogGroupName(lambdaName),
-      AWS_LAMBDA_LOG_STREAM_NAME: '2016/12/02/[$LATEST]f77ff5e4026c45bda9a9ebcec6bc9cad',
-      AWS_LAMBDA_FUNCTION_NAME: lambdaName,
-      AWS_LAMBDA_FUNCTION_MEMORY_SIZE: memorySize,
-      AWS_LAMBDA_FUNCTION_VERSION: '$LATEST',
-      NODE_PATH: '/var/runtime:/var/task:/var/runtime/node_modules',
-    };
+    return BbPromise.try(() => {
+      const lambdaName = this.options.functionObj.name;
+      const memorySize =
+        Number(this.options.functionObj.memorySize) ||
+        Number(this.serverless.service.provider.memorySize) ||
+        1024;
+
+      const lambdaDefaultEnvVars = {
+        LANG: 'en_US.UTF-8',
+        LD_LIBRARY_PATH:
+          '/usr/local/lib64/node-v4.3.x/lib:/lib64:/usr/lib64:/var/runtime:/var/runtime/lib:/var/task:/var/task/lib', // eslint-disable-line max-len
+        LAMBDA_TASK_ROOT: '/var/task',
+        LAMBDA_RUNTIME_DIR: '/var/runtime',
+        AWS_REGION: this.provider.getRegion(),
+        AWS_DEFAULT_REGION: this.provider.getRegion(),
+        AWS_LAMBDA_LOG_GROUP_NAME: this.provider.naming.getLogGroupName(lambdaName),
+        AWS_LAMBDA_LOG_STREAM_NAME: '2016/12/02/[$LATEST]f77ff5e4026c45bda9a9ebcec6bc9cad',
+        AWS_LAMBDA_FUNCTION_NAME: lambdaName,
+        AWS_LAMBDA_FUNCTION_MEMORY_SIZE: memorySize,
+        AWS_LAMBDA_FUNCTION_VERSION: '$LATEST',
+        NODE_PATH: '/var/runtime:/var/task:/var/runtime/node_modules',
+      };
 
-    const credentialEnvVars = this.getCredentialEnvVars();
+      const credentialEnvVars = this.getCredentialEnvVars();
 
-    // profile override from config
-    const profileOverride = this.provider.getProfile();
-    if (profileOverride) {
-      lambdaDefaultEnvVars.AWS_PROFILE = profileOverride;
-    }
+      // profile override from config
+      const profileOverride = this.provider.getProfile();
+      if (profileOverride) {
+        lambdaDefaultEnvVars.AWS_PROFILE = profileOverride;
+      }
 
-    const configuredEnvVars = this.getConfiguredEnvVars();
+      const configuredEnvVars = this.getConfiguredEnvVars();
 
-    _.merge(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+      const promises = _.entries(configuredEnvVars).map(([name, value]) => {
+        return this.resolveCfFunctionValue(this.provider, value)
```

Comment:
not sure it is useful to make this function part of `AwsInvokeLocal`, WDYT ?

Reply:
I believe we should pass value to `resolveCfFunctionValue` only when we're sure it's a CF instruction.

API where it can already be resolved string feels not clean (it's definitely confusing)